### PR TITLE
Update mapped and willow mappings

### PR DESCRIPTION
**Pull request by connorjcantrell**
### What
Updated the Mapped to Willow mapping file with additional mappings.

### Why
Enhanced the mapping, achieving over a 96% coverage of the Mapped ontology.

### Tested
Ensured the validity of the DTMI format and confirmed the absence of duplicate InputDtmi entries.

### CODE REVIEW

File: Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json [hunk 1]
```diff
@@ -3985,28 +4690,29 @@
...
-      "IsIgnored": true,
-      "OutputDtmi": "dtmi:com:willowinc:SpaceCode;1"
+      "InputDtmi": "dtmi:org:brickschema:schema:Brick:Solar_Zenith_Angle_Sensor;1",
+      "OutputDtmi": "dtmi:com:willowinc:SolarZenithAngleSensor;1"
     },
     {
-      "InputDtmi": "dtmi:org:brickschema:schema:Brick:Space_Heater;1",
-      "OutputDtmi": "dtmi:com:willowinc:UnitHeater;1"
+      "InputDtmi": "dtmi:org:brickschema:schema:Brick:Space;1",
+      "OutputDtmi": "dtmi:com:willowinc:Space;1"
```
Comment (joebeernink) [inline Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json:4711]: This should be room, not space. Willow.Space is considered an abstract class, and should not be directly assigned to

File: Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json [hunk 2]
```diff
@@ -70,10 +82,26 @@
       "InputDtmi": "dtmi:mapped:core:Air_Flow_Status;1",
       "OutputDtmi": "dtmi:com:willowinc:State;1"
     },
+    {
+      "InputDtmi": "dtmi:mapped:core:Air_Quality_Sensing_Device;1",
+      "OutputDtmi": "dtmi:com:willowinc:Device;1"
```
Comment (imercerwillow) [inline Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json:87]: Device seems rather generic. We have dtmi:com:willowinc:SensorEquipment;1 or what about dtmi:com:willowinc:IAQSensorEquipment;1 ?

File: Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json [hunk 3]
```diff
@@ -139,17 +203,49 @@
...
+      "InputDtmi": "dtmi:mapped:core:Chilled_Water_Usage_Sensor;1",
+      "OutputDtmi": "dtmi:com:willowinc:Sensor;1"
+    },
+    {
+      "InputDtmi": "dtmi:mapped:core:Class;1",
+      "OutputDtmi": "dtmi:com:willowinc:Class;1"
+    },
+    {
+      "InputDtmi": "dtmi:mapped:core:Close_Command;1",
+      "OutputDtmi": "dtmi:com:willowinc:ValveCloseActuator;1"
```
Comment (imercerwillow) [inline Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json:219]: Do we know it's a valve and not a damper, say? dtmi:com:willowinc:CloseActuator;1

image

File: Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json [hunk 4]
```diff
@@ -139,17 +203,49 @@
...
+    },
+    {
+      "InputDtmi": "dtmi:mapped:core:Close_Status;1",
+      "OutputDtmi": "dtmi:com:willowinc:PhaseBExportReactiveElectricalEnergySensor;1"
```
Comment (imercerwillow) [inline Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json:223]: Surely not! Maybe a CloseState?

image

File: Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json [hunk 5]
```diff
@@ -139,17 +203,49 @@
...
+      "InputDtmi": "dtmi:mapped:core:Coefficient_Of_Performance_Status;1",
+      "OutputDtmi": "dtmi:com:willowinc:CoefficientOfPerformance;1"
     },
     {
       "InputDtmi": "dtmi:mapped:core:Compressor_Status;1",
       "OutputDtmi": "dtmi:com:willowinc:State;1"
     },
+    {
+      "InputDtmi": "dtmi:mapped:core:Condenser_Water_Differential_Pressure_Sensor;1",
+      "OutputDtmi": "dtmi:com:willowinc:LeavingCondenserWaterPressureSensor;1"
```
Comment (imercerwillow) [inline Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json:247]: Is that a delta sensor? or should this be a CondenserWaterDeltaPressureSensor?

image

File: Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json [hunk 6]
```diff
@@ -166,6 +262,18 @@
       "InputDtmi": "dtmi:mapped:core:Condenser_Water_Supply_Temperature_Setpoint;1",
       "OutputDtmi": "dtmi:com:willowinc:EnteringCondenserWaterTemperatureSetpoint;1"
     },
+    {
+      "InputDtmi": "dtmi:mapped:core:Connected_Access_Point_Status;1",
+      "OutputDtmi": "dtmi:com:willowinc:State;1"
```
Comment (imercerwillow) [inline Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json:267]: dtmi:com:willowinc:OperationalState;1 ?

File: Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json [hunk 7]
```diff
@@ -166,6 +262,18 @@
...
+    },
+    {
+      "InputDtmi": "dtmi:mapped:core:Connected_Devices_Count_Status;1",
+      "OutputDtmi": "dtmi:com:willowinc:State;1"
+    },
+    {
+      "InputDtmi": "dtmi:mapped:core:Controller;1",
+      "OutputDtmi": "dtmi:com:willowinc:MotorController;1"
```
Comment (imercerwillow) [inline Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json:275]: Equipment;1 if we have no idea what kind of controller this is? Shouldn't guess Motor without any evidence.

File: Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json [hunk 8]
```diff
@@ -600,13 +1046,57 @@
...
+      "InputDtmi": "dtmi:mapped:core:Speaker;1",
+      "OutputDtmi": "dtmi:com:willowinc:Speaker;1"
+    },
+    {
+      "InputDtmi": "dtmi:mapped:core:Stage_Riser;1",
+      "OutputDtmi": "dtmi:com:willowinc:StageRiser;1"
+    },
+    {
+      "InputDtmi": "dtmi:mapped:core:Standby_Air_Temperature_Setpoint;1",
+      "OutputDtmi": "dtmi:com:willowinc:Setpoint;1"
```
Comment (imercerwillow) [inline Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json:1063]: dtmi:com:willowinc:AirTemperatureSetpoint;1

**Comment by connorjcantrell** [comment]
@imercerwillow Thanks for you input on these. I will make changes accordingly

File: Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json (hunk 4 above)
Comment (connorjcantrell) [reply Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json:223]: Yeah, this was definitely a mistake. Thanks for catching this

File: Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json (hunk 7 above)
Comment (connorjcantrell) [reply Ontologies.Mappings/src/Mappings/v1/Willow/mapped_v1_dtdlv2_Willow.json:275]: Agreed that Motor is not a fair assumption. Mapped introduced this concept to represent a Controller, which controls equipment. Granted, this concept is not currently utilized in practice. I will map this to Equipment for now.

**Comment by connorjcantrell** [comment]
@jbkoh This PR can't be merged until we update our ontology with Brick v1.3

File: Ontologies.Mappings/src/Mappings/v1/Mapped/willow_v1_dtdlv2_mapped.json [hunk 9]
```diff
@@ -358,7 +358,7 @@
     },
     {
       "InputDtmi": "dtmi:com:willowinc:COAirQualitySensor;1",
-      "OutputDtmi": "dtmi:org:brickschema:schema:Brick:CO_Level_Sensor;1"
```
Comment (jbkoh) [inline Ontologies.Mappings/src/Mappings/v1/Mapped/willow_v1_dtdlv2_mapped.json:361]: do not use co sensor but use co level sensor

File: Ontologies.Mappings/src/Mappings/v1/Mapped/willow_v1_dtdlv2_mapped.json [hunk 10]
```diff
@@ -130,7 +130,7 @@
     },
     {
       "InputDtmi": "dtmi:com:willowinc:AirHandlingUnit;1",
-      "OutputDtmi": "dtmi:org:brickschema:schema:Brick:Air_Handling_Unit;1"
+      "OutputDtmi": "dtmi:org:brickschema:schema:Brick:AHU;1"
```
Comment (jbkoh) [inline Ontologies.Mappings/src/Mappings/v1/Mapped/willow_v1_dtdlv2_mapped.json:133]: stick to canonical ones. I will share the list separately.

File: Ontologies.Mappings/src/Mappings/v1/Mapped/willow_v1_dtdlv2_mapped.json [hunk 11]
```diff
@@ -530,7 +530,7 @@
     },
     {
       "InputDtmi": "dtmi:com:willowinc:CoilEnteringWaterTemperatureSensor;1",
-      "OutputDtmi": "dtmi:org:brickschema:schema:Brick:Entering_Water_Temperature_Sensor;1"
+      "OutputDtmi": "dtmi:mapped:core:Heat_Exchanger_Return_Water_Temperature_Sensor;1"
```
Comment (jbkoh) [inline Ontologies.Mappings/src/Mappings/v1/Mapped/willow_v1_dtdlv2_mapped.json:533]: The definition of Heat Exchanger is more limited than Coil so we should stick to the previous one.

File: Ontologies.Mappings/src/Mappings/v1/Mapped/willow_v1_dtdlv2_mapped.json [hunk 12]
```diff
@@ -1258,7 +1258,7 @@
     },
     {
       "InputDtmi": "dtmi:com:willowinc:Equipment;1",
-      "OutputDtmi": "dtmi:org:brickschema:schema:Brick:Thing;1"
+      "OutputDtmi": "dtmi:mapped:core:Thing;1"
```
Comment (jbkoh) [inline Ontologies.Mappings/src/Mappings/v1/Mapped/willow_v1_dtdlv2_mapped.json:1261]: mapped it to mapped:Device. (brick:Equipment == mapped:Device)

File: Ontologies.Mappings/src/Mappings/v1/Mapped/willow_v1_dtdlv2_mapped.json [hunk 13]
```diff
@@ -2126,7 +2126,7 @@
     },
     {
       "InputDtmi": "dtmi:com:willowinc:LoadingDockEquipment;1",
-      "OutputDtmi": "dtmi:mapped:core:Device;1"
```
Comment (jbkoh) [inline Ontologies.Mappings/src/Mappings/v1/Mapped/willow_v1_dtdlv2_mapped.json:2129]: Device looks good to me.

**Review by jbkoh — CHANGES_REQUESTED** [review]
btw, map
willow:Space -> mapped:Place.
mapped:Place -> willow:Space.

Just realized that willow:Space is the top level concept.

**Review by jbkoh — APPROVED** [review]

**Pull request merged**